x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/TokenService.java

```diff
@@ -732,40 +787,72 @@ private void innerRefresh(String tokenDocId, Authentication userAuth, ActionList
+                                    ((Map<String, Object>) source.get("access_token")).get("user_token");
+                                final String authString = (String) userTokenSource.get("authentication");
+                                final Integer version = (Integer) userTokenSource.get("version");
+                                final Map<String, Object> metadata = (Map<String, Object>) userTokenSource.get("metadata");
+                                Version authVersion = Version.fromId(version);
+                                try (StreamInput in = StreamInput.wrap(Base64.getDecoder().decode(authString))) {
+                                    in.setVersion(authVersion);
+                                    Authentication authentication = new Authentication(in);
+                                    final String newUserTokenId = UUIDs.randomBase64UUID();
+                                    final Instant refreshTime = clock.instant();
+                                    Map<String, Object> updateMap = new HashMap<>();
+                                    updateMap.put("refreshed", true);
+                                    updateMap.put("refresh_time", refreshTime.toEpochMilli());
+                                    updateMap.put("superseded_by", getTokenDocumentId(newUserTokenId));
+                                    UpdateRequestBuilder updateRequest =
+                                        client.prepareUpdate(SecurityIndexManager.SECURITY_INDEX_NAME, TYPE, tokenDocId)
+                                            .setDoc("refresh_token", updateMap)
+                                            .setRefreshPolicy(RefreshPolicy.WAIT_UNTIL);
+                                    updateRequest.setIfSeqNo(response.getSeqNo());
+                                    updateRequest.setIfPrimaryTerm(response.getPrimaryTerm());
+                                    executeAsyncWithOrigin(client.threadPool().getThreadContext(), SECURITY_ORIGIN, updateRequest.request(),
+                                        ActionListener.<UpdateResponse>wrap(
+                                            updateResponse ->
+                                                createUserToken(newUserTokenId, authentication, userAuth, listener, metadata, true),
+                                            e -> {
+                                                Throwable cause = ExceptionsHelper.unwrapCause(e);
+                                                if (cause instanceof VersionConflictEngineException ||
+                                                    isShardNotAvailableException(e)) {
+                                                    innerRefresh(tokenDocId, userAuth,
+                                                        listener, attemptCount);
+                                                } else {
+                                                    onFailure.accept(e);
```

Comment:
This was not changed as part of this PR and we do handle the logging in the `onFailure` consumer that Tim added recently. I don't know if we gain something by printing these on `INFO` or `ERROR` here as we would print the Exception message also since the request will be failed. WDYT ?